units/src/amount/unsigned.rs

```diff
@@ -15,25 +15,28 @@ use super::{
     parse_signed_to_satoshi, split_amount_and_denomination, Denomination, Display, DisplayStyle,
     OutOfRangeError, ParseAmountError, ParseError, SignedAmount,
 };
-#[cfg(feature = "alloc")]
-use crate::{FeeRate, Weight};
 
-/// An amount.
+/// An unsigned bitcoin amount.
+///
+/// The [`Amount`] type can be used to express non-negative bitcoin amounts less than [`MAX_MONEY`].
+///
+/// > /** No amount larger than this is valid.
+/// > *
+/// > * Note that this constant is *not* the total money supply, which in Bitcoin
+/// > * currently happens to be less than 21,000,000 BTC for various reasons, but
+/// > * rather a sanity check.
 ///
-/// The [`Amount`] type can be used to express Bitcoin amounts that support arithmetic and
-/// conversion to various denominations. The [`Amount`] type does not implement [`serde`] traits
-/// but we do provide modules for serializing as satoshis or bitcoin.
+/// This type cannot be used to sum arbitrary amounts that may exceed `MAX_MONEY` e.g., the sum of
```

Comment:
Again, I don't see why it's worth throwing around the term MAX_MONEY.  What does the sum of transactions over some time period have to do with it?  Can it just be said that the sum can't exceed the total supply of Bitcoins?